src/librustc_trans/back/link.rs

```diff
@@ -676,8 +679,12 @@ fn link_natively(sess: &Session,
     } else {
         &sess.target.target.options.pre_link_objects_dll
     };
-    for obj in pre_link_objects {
-        cmd.arg(root.join(obj));
+    // When dynamically linking to MUSL we don't need to pass our startup
+    // objects as those will be provided by the MUSL toolchain
+    if !musl || crt_static {
```

Comment:
> Does anything leap to mind for you as an easy way to do that?

`pre_link_objects_crt_static` and `pre_link_objects_crt_dynamic`? 😄

BTW, is `pre_link_args` "setable" from JSON files? This field feels like a Rust distribution implementation detail because the objects get prepended with the sysroot's absolute path. I doubt custom targets will be able to use this field lest they modify the sysroot ...